### ISSUE TITLE
Graal EE produces more box nodes than CE

### ISSUE THREAD

**Issue by 2767mr** [oracle/graal#4867]
**Describe the issue**
Graal EE produces more box nodes than Graal CE

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. `git clone https://github.com/2767mr/graaljs`
2. `git checkout repro-2`
3. `cd graaljs/graal-js`
4. `mx sforceimports`
5. `mx build`
6. `mx --dynamicimports /compiler,/tools --jdk jvmci js --js.ecmascript-version=staging --vm.Dgraal.Dump=Truffle --vm.Dgraal.PrintGraph=Network" test.js`
7. Copy graaljs.jar into Graal EE.
8. `js --jvm --js.ecmascript-version=staging --vm.Dgraal.Dump=Truffle --vm.Dgraal.PrintGraph=Network"`

**Describe GraalVM and your environment:**
 - GraalVM version: EE 22.2.0, CE 22.3-dev-f9726d7 (also happens on 22.2 release tag)
 - JDK major version: 11
 - OS: Ubuntu 
 - Architecture: AMD64

**More details**
With CE:
![unknown](https://user-images.githubusercontent.com/2082553/188155295-ab34f803-d4a4-4710-aeb2-639ab9407b50.png)

With EE:
![unknown-1](https://user-images.githubusercontent.com/2082553/188155402-2007a462-7fa9-4bdc-915a-5b550d80a0b7.png)

See also: https://graalvm.slack.com/archives/CMY62M6FM/p1662119084319869?thread_ts=1661772999.382559&cid=CMY62M6FM

**Comment by gergo-** [comment]
@davleopo besides the box/unbox issue, shouldn't `IfNode.splitIfAtPhi` and/or duplication be able to eliminate these if nodes?

**Comment by davleopo** [comment]
yes, in theory it should. Depends a lot on the frequencies involved. We will have a closer look.

**Comment by gergo-** [comment]
Tracking this internally as GR-40908.
